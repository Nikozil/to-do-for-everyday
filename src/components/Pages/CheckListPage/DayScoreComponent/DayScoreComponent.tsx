import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addScore, addTag, Score } from '../../../../Redux/modules/tasksSlice';
import { AppStateType } from '../../../../Redux/store';
import ChangeTagForm from '../../../Forms/ChangeTagForm/ChangeTagForm';
import ChangeScoreForm from '../../../Forms/ChangeScoreForm/ChangeScoreForm';

const DayScoreComponent = () => {
  const score = useSelector((state: AppStateType) => state.tasks.doneDay.score);
  const tag = useSelector((state: AppStateType) => state.tasks.doneDay.tag);
  const dispatch = useDispatch();

  const addTagHandler = (tag: string) => {
    dispatch(addTag(tag));
  };
  const addScoreHandler = (score: Score) => {
    dispatch(addScore(score));
  };
  return (
    <div>
      Итоги дня
      <ChangeTagForm tag={tag} handleSubmit={addTagHandler} />
      <ChangeScoreForm score={score} handleSubmit={addScoreHandler} />
    </div>
  );
};

export default DayScoreComponent;
