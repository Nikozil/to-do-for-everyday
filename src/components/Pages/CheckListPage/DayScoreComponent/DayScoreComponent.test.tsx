import React from 'react';
import { render } from '../../../../utils/tests/test-utils';
import DayScoreComponent from './DayScoreComponent';

describe('DayScoreComponent', () => {
  it('render correct', async () => {
    const { asFragment } = render(<DayScoreComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
