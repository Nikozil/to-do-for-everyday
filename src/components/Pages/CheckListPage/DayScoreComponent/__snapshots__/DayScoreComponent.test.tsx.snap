// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayScoreComponent render correct 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Итоги дня
    </h2>
    <div
      class="my-1"
    >
      <form
        action="#"
      >
        <div
          class="row g-2 align-items-center"
        >
          <div
            class="col-auto"
          >
            <label
              class="col-form-label-sm fw-bold"
              for="tag"
            >
              #Tag дня:
            </label>
          </div>
          <div
            class="col"
          >
            <textarea
              autocomplete="off"
              class="form-control form-control-sm textarea"
              id="tag"
              name="tag"
              placeholder="Tag"
              rows="1"
              type="text"
            />
          </div>
        </div>
      </form>
    </div>
    <div>
      <form
        action="#"
      >
        <div
          class="row g-2 align-items-center"
        >
          <div
            class="col-auto"
          >
            <div
              class="col-form-label-sm fw-bold"
              id="my-radio-group"
            >
              Оченка дня
            </div>
          </div>
          <div
            class="col-auto"
          >
            <div
              aria-labelledby="my-radio-group"
              role="group"
            >
              <label
                class="star"
              >
                <input
                  class="star__radio"
                  name="score"
                  type="radio"
                  value="1"
                />
                <svg
                  fill="currentColor"
                  height="29"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 16 16"
                  width="29"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
                  />
                </svg>
              </label>
              <label
                class="star"
              >
                <input
                  class="star__radio"
                  name="score"
                  type="radio"
                  value="2"
                />
                <svg
                  fill="currentColor"
                  height="29"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 16 16"
                  width="29"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
                  />
                </svg>
              </label>
              <label
                class="star"
              >
                <input
                  class="star__radio"
                  name="score"
                  type="radio"
                  value="3"
                />
                <svg
                  fill="currentColor"
                  height="29"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 16 16"
                  width="29"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
                  />
                </svg>
              </label>
              <label
                class="star"
              >
                <input
                  class="star__radio"
                  name="score"
                  type="radio"
                  value="4"
                />
                <svg
                  fill="currentColor"
                  height="29"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 16 16"
                  width="29"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
                  />
                </svg>
              </label>
              <label
                class="star"
              >
                <input
                  class="star__radio"
                  name="score"
                  type="radio"
                  value="5"
                />
                <svg
                  fill="currentColor"
                  height="29"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 16 16"
                  width="29"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
                  />
                </svg>
              </label>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
   
</DocumentFragment>
`;
